packages/rum/src/domain/record/record.ts

```diff
@@ -42,11 +42,15 @@ export function record(options: RecordOptions): RecordAPI {
     throw new Error('emit function is required')
   }
 
+  let isFullSnapshotPending = false
+
   const emitAndComputeStats = (record: BrowserRecord) => {
-    emit(record)
-    sendToExtension('record', { record })
-    const view = options.viewContexts.findView()!
-    replayStats.addRecord(view.id)
+    if (!isFullSnapshotPending) {
```

Comment:
**💬 suggestion:** ‏write one or multiple unit tests to make sure this works as expected. Based on [my comment](https://github.com/DataDog/browser-sdk/pull/2756#pullrequestreview-2122327947), this could look like:

```ts
it("ignores any record while a full snapshot is pending", () => {
  ...
```